tuf/client_rework/updater_rework.py

```diff
@@ -83,13 +79,18 @@ def refresh(self) -> None:
         self._load_snapshot()
         self._load_targets("targets", "root")
 
-    def get_one_valid_targetinfo(self, filename: str) -> Dict:
+    def get_one_valid_targetinfo(self, target_path: str) -> Dict:
         """
-        Returns the target information for a specific file identified by its
-        file path.  This target method also downloads the metadata of updated
-        targets.
+        Returns the target information for a target identified by target_path.
+
+        As a side-effect this method downloads all the metadata it needs to
+        return the target information.
+
+        Args:
+            target_path: A path-relative-URL string
```

Comment:
relative to?

Reply:
A fantastic question that I'd love to have an answer to... This unholy merger of `target identifier` and `relative local path` and `url fragment` is one of the sharp corners of TUF. For this function, target_path is just an identifier that's used to find the relevant pieces of metadata  ... but it is expected to somehow also transform nicely into a relative local filesystem path AND a part of a URL. Depending on the use case the user might be thinking in terms of URLs (like pip fetching a specific index.html) or local file paths (a system updater that wants to make sure the newest firmware file is on local disk).

In a way the url-fragment is not necessarily relative to anything as the hash-prefixing might mean $TARGETSBASEURL + $TARGETPATH isn't actually the url we want...  I'll try to think of a way to word this.

Reply:
A realization now that I wrote it down: Is the idea that target_paths are directly/uniquely transferable to paths on local filesystem actually from the spec at all, or is it just the way the reference client was written? Could we just as well have decided to download `http://localhost:8080/targets/a/b/c.txt` into just `c.txt` or asked the user to provide the local filename?

If this is purely a client decision, is it good one?

Reply:
> client MUST write the file to non-volatile storage as FILENAME.EXT

Looks like path handling is not in spec, filename is (even that might be just copy-paste from metadata handling rules where it makes sense)

Reply:
I've tried improving it a bit... not sure how to proceed other than making a not in #1385

Reply:
thanks for tackling this, cleaned up doctoring and adding a note to #1385 feels sufficient.